crates/ruff_linter/src/rules/flake8_unused_arguments/rules/unused_arguments.rs

```diff
@@ -311,6 +311,63 @@ fn call<'a>(
     }));
 }
 
+/// Returns `true` if a function appears to be a base class stub. In other
+/// words, if it matches the following syntax:
+///
+/// ```text
+/// variable = <string | f-string>
+/// raise NotImplementedError(variable)
+/// ```
+///
+/// See also [`is_stub`]. We're a bit more generous in what is considered a
+/// stub in this rule to avoid clashing with [`EM101`].
+///
+/// [`is_stub`]: function_type::is_stub
+/// [`EM101`]: https://docs.astral.sh/ruff/rules/raw-string-in-exception/
+fn is_not_implemented_stub_with_variable(function_def: &StmtFunctionDef) -> bool {
+    // Ignore doc-strings.
+    let statements = match function_def.body.as_slice() {
+        [Stmt::Expr(StmtExpr { value, .. }), rest @ ..]
+            if matches!(**value, ast::Expr::StringLiteral(_)) =>
+        {
+            rest
+        }
```

Comment:
nit: we could directly use the `is_string_literal_expr` method here
```suggestion
        [Stmt::Expr(StmtExpr { value, .. }), rest @ ..] if value.is_string_literal_expr() => rest,
```